### PR TITLE
fix warnings in tests

### LINKED ISSUE #621: errorprone warnings in tests
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/install/DownloaderTest.java:[173,17] [CatchFail] Ignoring exceptions and calling fail() is unnecessary, and makes test output less useful
    (see http://errorprone.info/bugpattern/CatchFail)
  Did you mean 'throw new AssertionError("Test failed due to IOException", e);'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java:[36,86] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
    (see http://errorprone.info/bugpattern/DefaultCharset)
  Did you mean 'private InputStream fakeInputStream = new ByteArrayInputStream(TEST_STRING.getBytes(UTF_8));' or 'private InputStream fakeInputStream = new ByteArrayInputStream(TEST_STRING.getBytes(Charset.defaultCharset()));'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java:[83,21] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
    (see http://errorprone.info/bugpattern/DefaultCharset)
  Did you mean 'result.append(new String(bytes.getAllValues().get(i), 0, nBytes.getAllValues().get(i), UTF_8));' or 'result.append(new String(bytes.getAllValues().get(i), 0, nBytes.getAllValues().get(i), Charset.defaultCharset()));'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/VersionTest.java:[39,7] [MissingFail] Not calling fail() when expecting an exception masks bugs
    (see http://errorprone.info/bugpattern/MissingFail)
  Did you mean 'new Version("1");'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/ExitCodeRecorderProcessExitListenerTest.java:[47,18] [BoxedPrimitiveConstructor] valueOf or autoboxing provides better time and space performance
    (see http://errorprone.info/bugpattern/BoxedPrimitiveConstructor)
  Did you mean 'assertEquals(Integer.valueOf(code), listener.getMostRecentExitCode());'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/libraries/CloudLibrariesTest.java:[39,11] [AssertionFailureIgnored] This assertion throws an AssertionError if it fails, which will be caught by an enclosing try block.
    (see http://errorprone.info/bugpattern/AssertionFailureIgnored)
  Did you mean 'AssertionError e = assertThrows(AssertionError.class, () -> cloudLibraries.getLibraries());'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAuthTest.java:[77,7] [CatchAndPrintStackTrace] Logging or rethrowing exceptions should usually be preferred to catching and calling printStackTrace
    (see http://errorprone.info/bugpattern/CatchAndPrintStackTrace)
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkTest.java:[72,5] [CatchFail] Ignoring exceptions and calling fail() is unnecessary, and makes test output less useful
    (see http://errorprone.info/bugpattern/CatchFail)
  Did you mean 'throw new AssertionError("Cloud SDK validation should not validate jdk", aee);' or 'public void testValidateCloudSdk_doesNotThrowInvalidJdkException() throws AppEngineException{'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java:[29,41] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
    (see http://errorprone.info/bugpattern/DefaultCharset)
  Did you mean 'testHandler.bytes("line1\n".getBytes(UTF_8), 6);' or 'testHandler.bytes("line1\n".getBytes(Charset.defaultCharset()), 6);'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java:[30,41] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
    (see http://errorprone.info/bugpattern/DefaultCharset)
  Did you mean 'testHandler.bytes("line2\n".getBytes(UTF_8), 6);' or 'testHandler.bytes("line2\n".getBytes(Charset.defaultCharset()), 6);'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java:[37,44] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.
    (see http://errorprone.info/bugpattern/DefaultCharset)
  Did you mean 'testHandler.bytes("line1\n123".getBytes(UTF_8), 6);' or 'testHandler.bytes("line1\n123".getBytes(Charset.defaultCharset()), 6);'?
[WARNING] /Users/elharo/appengine-plugins-core/src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java:[38,44] [DefaultCharset] Implicit use of the platform default charset, which can result in differing behaviour between JVM executions or incorrect behavior if the encoding of the data source doesn't match expectations.

### PR DESCRIPTION
**Pull request by elharo**
@loosebazooka fix #621

### CODE REVIEW

File: src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java [hunk 1]
```diff
@@ -21,12 +21,14 @@
 import com.google.gson.Gson;
 import com.google.gson.reflect.TypeToken;
 import com.google.gson.stream.JsonReader;
+
```
Comment (chanseokoh) [inline src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java]: This should be reverted per Google style. Hope you don't get a warning.

File: src/test/java/com/google/cloud/tools/libraries/CloudLibrariesTest.java [hunk 2]
```diff
@@ -21,6 +21,8 @@
 import static org.junit.Assert.fail;
 
 import java.io.IOException;
+import java.util.MissingResourceException;
+
```
Comment (chanseokoh) [inline src/test/java/com/google/cloud/tools/libraries/CloudLibrariesTest.java]: extra line

File: src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java [hunk 3]
```diff
@@ -33,7 +34,8 @@
 public class AsyncByteConsumerTest {
 
   private static final String TEST_STRING = "test line1\ntest line2\n";
-  private InputStream fakeInputStream = new ByteArrayInputStream(TEST_STRING.getBytes());
+  private InputStream fakeInputStream =
```
Comment (chanseokoh) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java]: `final` if you like it.

File: src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java [hunk 4]
```diff
@@ -17,6 +17,9 @@
 package com.google.cloud.tools.managedcloudsdk.command;
 
 import com.google.cloud.tools.managedcloudsdk.command.AsyncStreamSaverFactory.CollectingByteHandler;
+
```
Comment (chanseokoh) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java]: extra blank

File: src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java (hunk 1 above)
Comment (elharo) [reply src/main/java/com/google/cloud/tools/libraries/CloudLibraries.java]: done

File: src/test/java/com/google/cloud/tools/libraries/CloudLibrariesTest.java (hunk 2 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/libraries/CloudLibrariesTest.java]: done

File: src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java (hunk 3 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/command/AsyncByteConsumerTest.java]: done

File: src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java (hunk 4 above)
Comment (elharo) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java]: done

Comment (chanseokoh) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/command/CollectingByteHandlerTest.java]: This one is still missing.

File: src/test/java/com/google/cloud/tools/managedcloudsdk/install/DownloaderTest.java [hunk 5]
```diff
@@ -175,8 +175,8 @@ public void run() {
                   Assert.fail("InterruptedException expected but not thrown.");
                 } catch (InterruptedException ex) {
                   Assert.assertEquals("Download was interrupted", ex.getMessage());
-                } catch (IOException e) {
-                  Assert.fail("Test failed due to IOException");
+                } catch (IOException ex) {
+                  throw new AssertionError("Test failed due to IOException", ex);
```
Comment (chanseokoh) [inline src/test/java/com/google/cloud/tools/managedcloudsdk/install/DownloaderTest.java]: I actually see a bigger problem in this test. Failing in a different thread won't fail the actual test, AFAIK. We need to propagate the result to the main test thread instead. It'll be a bit involved (not so much). Let me file an issue and tackle this myself.

Comment (loosebazooka) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/install/DownloaderTest.java]: Yeah, it appears like I missed something with this test. The fix is to use an executor service and Callable, and the exception will be propagated.

Comment (chanseokoh) [reply src/test/java/com/google/cloud/tools/managedcloudsdk/install/DownloaderTest.java]: Ah, good to know. I was trying a different approach, but the executor services seems nice.

File: src/main/java/com/google/cloud/tools/managedcloudsdk/command/AsyncStreamSaverFactory.java [hunk 6]
```diff
@@ -18,6 +18,8 @@
 
 import java.nio.charset.StandardCharsets;
```
[the remark this answers is not in the document]
Comment (elharo) [reply src/main/java/com/google/cloud/tools/managedcloudsdk/command/AsyncStreamSaverFactory.java]: I should check if there'a a new Eclipse plugin to handle the updated style.

**Pull request merged**